language/move-prover/boogie-backend/src/bytecode_translator.rs

```diff
@@ -1474,137 +1854,221 @@ impl<'env> FunctionTranslator<'env> {
                             str_local(dest),
+                            mod_type,
                             str_local(op1),
                             str_local(op2)
                         );
                     }
-                    Gt => {
+                    Shl | Shr => {
                         let dest = dests[0];
                         let op1 = srcs[0];
                         let op2 = srcs[1];
-                        emitln!(
-                            writer,
-                            "call {} := $Gt({}, {});",
-                            str_local(dest),
-                            str_local(op1),
-                            str_local(op2)
-                        );
-                    }
-                    Le => {
-                        let dest = dests[0];
-                        let op1 = srcs[0];
-                        let op2 = srcs[1];
-                        emitln!(
-                            writer,
-                            "call {} := $Le({}, {});",
-                            str_local(dest),
-                            str_local(op1),
-                            str_local(op2)
-                        );
+                        let sh_oper_str = if oper == &Shl { "Shl" } else { "Shr" };
+                        let num_oper = global_state
+                            .get_temp_index_oper(mid, fid, dest, baseline_flag)
+                            .unwrap();
+                        let bv_flag = self.bv_flag(num_oper);
+                        if bv_flag {
+                            let target_type = match &self.get_local_type(dest) {
+                                Type::Primitive(PrimitiveType::U8) => "Bv8",
+                                Type::Primitive(PrimitiveType::U16) => "Bv16",
+                                Type::Primitive(PrimitiveType::U32) => "Bv32",
+                                Type::Primitive(PrimitiveType::U64) => "Bv64",
+                                Type::Primitive(PrimitiveType::U128) => "Bv128",
+                                Type::Primitive(PrimitiveType::U256) => "Bv256",
+                                Type::Primitive(_)
+                                | Type::Tuple(_)
+                                | Type::Vector(_)
+                                | Type::Struct(_, _, _)
+                                | Type::TypeParameter(_)
+                                | Type::Reference(_, _)
+                                | Type::Fun(_, _)
+                                | Type::TypeDomain(_)
+                                | Type::ResourceDomain(_, _, _)
+                                | Type::Error
+                                | Type::Var(_) => unreachable!(),
+                            };
+                            let src_type = match &self.get_local_type(op2) {
+                                Type::Primitive(PrimitiveType::U8) => "8",
+                                Type::Primitive(PrimitiveType::U16) => "16",
```

Comment:
Factor this out in an independent function and reuse it?

E.g. `fun get_bit_with(ty: &Type) -> String` and then you can use this in `format!("Bv{}", get_bit_width(ty))`